File: 2022/day4/solution.rb
#!/usr/bin/ruby
input = File.read("input").split

total1 = 0
total2 = 0
for a in input
  b = a.split(/,|-/).map! {|x| x = Integer(x)}
  if (b[0] <= b[2] and b[1] >= b[3]) or (b[0] >= b[2] and b[1] <= b[3])
    total1 += 1
  end
  if (b[0]..b[1]).include? b[2] or (b[0]..b[1]).include? b[3] or (b[2]..b[3]).include? b[0] or (b[2]..b[3]).include? b[1]
    total2 += 1
  end
end

print "Part 1: ", total1, "\n"
print "Part 2: ", total2, "\n"

File: 2022/day4/solution.py
#!/usr/bin/python3
import re
total1, total2 = 0, 0
with open("input", "r") as file:
    for i in file:
       i = list(map(int, re.findall(r'\d+', i)))
       total1 += (i[0] <= i[2] and i[1] >= i[3]) or (i[0] >= i[2] and i[1] <= i[3])
       total2 += bool(set(range(i[0], i[1] + 1)) & set(range(i[2], i[3] + 1)))
print(f"Part 1: {total1}\nPart 2: {total2}")

